kornia/image/image.py

```diff
@@ -0,0 +1,254 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Any
+
+import torch
+from torch.utils.dlpack import from_dlpack, to_dlpack
+
+from kornia.core import Device, Dtype, Tensor
+from kornia.core.check import KORNIA_CHECK_SHAPE
+from kornia.image.base import ChannelsOrder, ImageLayout, ImageSize, PixelFormat
+from kornia.io.io import ImageLoadType, load_image, write_image
+
+# placeholder for numpy
+np_ndarray = Any
+DLPack = Any
+
+
+class Image:
+    r"""Class that holds an Image Tensor representation.
+
+    .. note::
+
+        Disclaimer: This class provides the minimum functionality for image manipulation. However, as soon
+        as you start to experiment with advanced tensor manipulation, you might expect fancy
+        polymorphic behaviours.
+
+    .. warning::
+
+        This API is experimental and might suffer changes in the future.
+
+    Args:
+        data: a torch tensor containing the image data.
+        layout: a dataclass containing the image layout information.
+
+    Examples:
+        >>> # from a torch.tensor
+        >>> data = torch.randint(0, 255, (3, 4, 5))  # CxHxW
+        >>> layout = ImageLayout(
+        ...     image_size=ImageSize(4, 5),
+        ...     channels=3,
+        ...     pixel_format=PixelFormat.RGB,
+        ...     channels_order=ChannelsOrder.CHANNELS_FIRST,
+        ... )
+        >>> img = Image(data, layout)
+        >>> assert img.channels == 3
+
+        >>> # from a numpy array (like opencv)
+        >>> data = np.ones((4, 5, 3), dtype=np.uint8)  # HxWxC
+        >>> img = Image.from_numpy(data, pixel_format=PixelFormat.BGR)
+        >>> assert img.channels == 3
+        >>> assert img.width == 5
+        >>> assert img.height == 4
+    """
+
+    def __init__(self, data: Tensor, layout: ImageLayout) -> None:
+        """Image constructor.
+
+        Args:
+            data: a torch tensor containing the image data.
+            layout: a dataclass containing the image layout information.
+        """
+        # TODO: move this to a function KORNIA_CHECK_IMAGE_LAYOUT
+        if layout.channels_order == ChannelsOrder.CHANNELS_FIRST:
+            shape = [str(layout.channels), str(layout.image_size.height), str(layout.image_size.width)]
+        elif layout.channels_order == ChannelsOrder.CHANNELS_LAST:
+            shape = [str(layout.image_size.height), str(layout.image_size.width), str(layout.channels)]
+        else:
+            raise NotImplementedError(f"Layout {layout.channels_order} not implemented.")
+
+        KORNIA_CHECK_SHAPE(data, shape)
```

Comment:
Maybe should work with [KORNIA_CHECK_IS_IMAGE](https://github.com/kornia/kornia/blob/master/kornia/core/check.py#L343) here

Reply:
sure but i'd prefer to modify that one after we get this api more mature